### DISCUSSION TITLE
About design-system components.

### DISCUSSION THREAD

**Discussion by alantoa** [showtime-xyz/showtime-frontend#963] [Show and tell]
design-system components  add some extra props.

The current components are all using `children` props, would result in having to use double `<blockquote>`.

Maybe we can add a props to use single tag.

Just like, 

Before:

``` jsx
  <Text>Text</Text>

  <Button>Title</Button>
```

After :

``` jsx
  <Text tx="Text" />

  <Button title="Title" />
```

I think it can make the code cleaner, what do you think?

**Comment by axeldelafosse** [comment]
> I think it can make the code cleaner, what do you think?
 
I personally don't think that it looks cleaner. It feels more natural to use children (like you would with bare HTML)

**Reply by alantoa** [reply]
Alright, I think JSX should be more flexible than HTML and do it with less code, JSX could support both methods~ 😄

**Comment by alantoa** [comment]
![image](https://user-images.githubusercontent.com/37520667/162921986-9cfb85eb-5054-440d-8618-f56b5d1895a9.png)

**Comment by enesozturk** [comment]
They are both the same for me as readiblity. I think this should depend on other features of the Text component (or Button). Maybe we should consider these also for future.

Here is an example: When we want to enable the multi-lang feature, we will probably go in two ways;
 - Call the translate module in Text comp and pass `translateId` (or something like that) prop to it. In this way, we don't need to import i18 in every component we use Text. Big plus.
```js
<Text translateId="components.info.description" />
```
 - Passing translated string as children to Text. In this usage, we need to import i18n in all the components we use Text.
```js
<Text>
  {I18n.t("components.info.description")}
</Text>
```
If we don't handle these or some other features, they are almost the same for me.

**Reply by alantoa** [reply]
Yeah, I agree!
